document-generator-core/swagger-2.0/spec/document-generator.json

```diff
@@ -0,0 +1,57 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Document generator API",
+    "description": "An API for generation of documents",
+    "version": "1.0.0"
+  },
+  "host": "localhost",
+  "schemes": [
+    "http"
+  ],
+  "tags": [
+    {
+    "name": "document-generator"
+    }
+  ],
+  "paths": {
+    "/document-generate": {
+      "post": {
+        "x-operationName": "generate",
+        "summary": "Generation of documents",
+        "description": "Generate a document using the request to obtain the document data and send to render service",
+        "tags": [
+          "document-generator"
+        ],
+        "consumes": [
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "name": "data",
+            "in": "body",
+            "description": "required parameters to generate a document",
+            "required": true,
+            "schema": {
+              "$ref": "http://localhost:3123/swagger-2.0/models/document-generator.json#/definitions/documentGeneratorRequest"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "The document has been created",
+            "schema": {
+              "$ref": "http://localhost:3123/swagger-2.0/models/document-generator.json#/definitions/documentGeneratorResponse"
+            }
+          },
+          "400": {
+            "description": "Unable to handle given request"
```

Comment:
why? what conditions would result in this error status? expliciti examples can be seen in responses here http://ch-jenkins-rh1.internal.ch:3123/manipulate-company-data/reference/transactions/update

Reply:
I was told to return 400 which as far As I was aware had been discussed. 400, 401 to be returned. If this needs to change, it can be done. But I think a clear outline of what should be returned needs to be agreed

Reply:
Rewording the initial comment - the description should inform the reader the conditions that lead to a 400 response